EBH-3338/medicover.py

```
@@ -0,0 +1,98 @@
+for bam in list_of_bams:
+    pattern = r'SP[a-zA-Z0-9]+'
+    match = re.search(pattern, bam)
+    # If match not found, then try another pattern
+    if match is None:
+        pattern = r'[gG][mM][0-9]+_?[0-9]+'
+        match = re.search(pattern, bam)
+    
+    samplenames_bam.append(match.group())
+    # If match still not found, raise error
+    if match is None:
+        raise TypeError(f"The {bam} has no matched sample name")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)